src/library_addfunction.js

```diff
@@ -151,11 +151,13 @@ mergeInto(LibraryManager.library, {
 
   $updateTableMap__deps: ['$getWasmTableEntry'],
   $updateTableMap: function(offset, count) {
-    for (var i = offset; i < offset + count; i++) {
-      var item = getWasmTableEntry(i);
-      // Ignore null values.
-      if (item) {
-        functionsInTableMap.set(item, i);
+    if (functionsInTableMap) {
+      for (var i = offset; i < offset + count; i++) {
+        var item = getWasmTableEntry(i);
+        // Ignore null values.
+        if (item) {
+          functionsInTableMap.set(item, i);
+        }
```

Comment:
Is this a separate fix?

Reply:
This is actually needed work around a JSDCE optmizer + closure issue.

Here is how I remember it going (roughly):

1. addFunction (which depends on updateTableMap + functionsInTableMap) get added due to `LEGACY_RUNTIME` being set by default.
2. JSDCE run which is able to remove addFunction.. but it only run a single round so leaved the now used `updateTableMap` function in place.
3. Closure compiler then run as sees that there is no place where `functionsInTableMap` is ever set to anything by `undefined` which means it can then prove that `functionsInTableMap.set` is invalid here.

This seems like a bug that can occur when JSDCE (the single pass version) is used with closure compiler.. but its probably fairly rare. 

In any case this seems like a good fix.  If `functionsInTableMap` has not yet been initilzed when `updateTableMap` is called it would crash (without this fix).

Reply:
What would closure actually do in step 3? It seems like any change it makes is ok since it's to a function that is never called.

Regardless I think you are right, reading the code it looks like this has been a possible bug. To hit we'd need to load a side module that somehow avoids calling `addFunction`  at all, and then the dylink call to `updateTableMap` will throw.

Reply:
closure reports an error saying `functionsInTableMap.set` is not a function since it knows that `functionsInTableMap` is undefined.  By adding the check here make closure happy since it can tell that the `undefined` is being checked for.

Reply:
I see, thanks. Yes, I guess partially optimized code can end up hitting closure errors in that way.